graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH5.kt

```diff
@@ -0,0 +1,126 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphFunctionalElements
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.GetShortestPath, GraphFunctionalElements.GraphFunction, """
+
+""")
+class BEH5 : GraphScenario() {
+
+    @Eac("""
+        Given a graph that contains at least path from a beginning vertex to an ending vertex,
+        the graph will have a shortest path between these vertices
+    """)
+    fun onePath() {
+        given {
+            thereIsAnUndirectedGraph(graphA) {
+                edge("a", "b")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "b")
+        }
+    }
+
+    @Eac("""
+        Given a graph that contains more then one path between two vertices,
+        there is a shortest path
+    """, """
+        If two paths have the same number of vertices and are both the shortest only
+        the first path defined will be recognised as the shortest path.
+    """)
+    fun multiplePaths() {
+        given {
+            thereIsAnUndirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("c", "d")
+                edge("d", "e")
+                edge("a", "f")
+                edge("f", "e")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "f", "e")
+        }
+    }
+
+    @Eac("""
+        Given a directed graph,
+        the shortest path wont include an edge that is declared in the opposite direction to the path
+    """)
+    fun directedGraph() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("c", "d")
+                edge("a", "e")
+                edge("d", "e")
+            }
+        }
+        then {
+            hasShortestPath(graphA, "a", "b", "c", "d")
+        }
+    }
+
+    @Eac("""
+        Given a graph that has a path with a cycle, the cycle will not be included in the shortest path
+    """, """
+        Vertices from the cycle, that are not part of the actual shortest path, will not be included.
```

Comment:
🚫
```suggestion
         A shortest path cannot contain non-trivial cycles
    """, """
        Indeed, replacing a cycle with its base point results in another path between the same start and end vertices with strictly shorter length.
```
The "non-trivial" bit is important, because as we've seen before a single vertex can be seen as a cycle.